File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Web/Version2012/1033/AjaxServerControl/ServerControl1.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
$if$ ($targetframeworkversion$ >= 3.5)using System.Xml.Linq;
$endif$
namespace $safeprojectname$
{
    /// <summary>
    /// Summary description for ServerControl1
    /// </summary>
    public class ServerControl1 : ScriptControl
    {
        public ServerControl1()
        {
            //
            // TODO: Add constructor logic here
            //
        }
        protected override IEnumerable<ScriptDescriptor>
                GetScriptDescriptors()
        {
            ScriptControlDescriptor descriptor = new ScriptControlDescriptor("$safeprojectname$.ClientControl1", this.ClientID);
            yield return descriptor;
        }

        // Generate the script reference
        protected override IEnumerable<ScriptReference>
                GetScriptReferences()
        {
            yield return new ScriptReference("$safeprojectname$.ClientControl1.js", this.GetType().Assembly.FullName);
        }
    }
}

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Web/Version2012/1033/AjaxServerControl/ServerControl1.vb
Public Class ServerControl1
    Inherits ScriptControl

    Protected Overrides Function GetScriptDescriptors() As IEnumerable(Of ScriptDescriptor)
        Dim descriptor As New ScriptControlDescriptor("$safeprojectname$.ClientControl1", Me.ClientID)

        Return New List(Of ScriptDescriptor) From {descriptor}
    End Function

    ' Generate the script reference
    Protected Overrides Function GetScriptReferences() As IEnumerable(Of ScriptReference)
        Dim scriptRef As New ScriptReference("$safeprojectname$.ClientControl1.js", Me.GetType().Assembly.FullName)

        Return New List(Of ScriptReference) From {scriptRef}
    End Function
End Class
